src/app/customizable-demo-form.component.ts

```diff
@@ -195,13 +271,34 @@ export class CustomizableDemoForm {
     this.sliderScoreThresholds[2] = (1 + this.sliderScoreThresholds[1]) / 2;
   }
 
+  onDemoSettingsChanged() {
+    let newDemoSettings = this.getDemoSettings();
+    if (JSON.stringify(this.demoSettings) !== JSON.stringify(newDemoSettings)) {
```

Comment:
This isn't quite a perfect check of whether they're equal, if you're relying on it being true for anything, you should switch to something like _.isEqual.

On another note, I think this would look cleaner as an early-return if they are equal and then having the case for them being equal at the main level in the function.

Reply:
I don't think there is a guarantee that JSON.stringify produces a consistent sort order for the elements; so this kind of check is generally not safe. +1 to using something like _.isEqual

Reply:
Done (commit not pushed yet, addressing each comment one at a time)